tests/testsuite/lockfile_path.rs

```diff
@@ -0,0 +1,510 @@
+    let lockfile_path_argument = "mylockfile/is/burried/Cargo.lock";
+    let p = make_remove_project().build();
+    run_remove_command(&p, lockfile_path_argument.to_string());
+
+    assert!(!p.root().join("Cargo.lock").exists());
+    assert!(p.root().join(lockfile_path_argument).is_file());
+}
+
+#[cargo_test]
+fn remove_lockfile_read() {
+    let lockfile_path_argument = "mylockfile/Cargo.lock";
+    let p = make_remove_project()
+        .file("mylockfile/Cargo.lock", VALID_LOCKFILE)
+        .build();
+    run_remove_command(&p, lockfile_path_argument.to_string());
+
+    assert!(!p.root().join("Cargo.lock").exists());
+    assert!(p.root().join(lockfile_path_argument).is_file());
+}
+
+#[cargo_test]
+fn remove_lockfile_override() {
+    let lockfile_path_argument = "mylockfile/Cargo.lock";
+    let p = make_remove_project()
+        .file("Cargo.lock", "This is an invalid lock file!")
+        .build();
+    run_remove_command(&p, lockfile_path_argument.to_string());
+
+    assert!(p.root().join(lockfile_path_argument).is_file());
+}
+
+#[cargo_test]
+fn assert_respect_pinned_version_from_lockfile_path() {
+    const PINNED_VERSION: &str = r#"# This file is automatically @generated by Cargo.
+# It is not intended for manual editing.
+version = 3
+
+[[package]]
+name = "hello"
+version = "1.0.2"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "2b94f16c310ebbd9efcca5a5a17131d70bd454876f2af007f3da211afadff4fc"
+
+[[package]]
+name = "test_foo"
+version = "0.5.0"
+dependencies = [
+ "hello",
+]
+"#;
+    const TOML: &str = r#"#
+[package]
+
+name = "test_foo"
+version = "0.5.0"
+authors = ["wycats@example.com"]
+edition = "2015"
+
+[[bin]]
+
+name = "test_foo"
+
+[dependencies]
+hello = "1.0.0"
+"#;
+
+    let lockfile_path_argument = "mylockfile/Cargo.lock";
+    let p = project()
+        .file("Cargo.toml", TOML)
+        .file("src/main.rs", "fn main() {}")
+        .file("mylockfile/Cargo.lock", PINNED_VERSION)
+        .build();
+
+    make_execs(&mut p.cargo("package"), lockfile_path_argument.to_string()).run();
+
+    assert!(!p.root().join("Cargo.lock").exists());
+    assert!(p.root().join(lockfile_path_argument).is_file());
+
+    assert!(p
+        .root()
+        .join("target/package/test_foo-0.5.0/Cargo.lock")
+        .is_file());
+
+    let path = p.root().join("target/package/test_foo-0.5.0/Cargo.lock");
+    let contents = fs::read_to_string(path).unwrap();
+
+    assert_e2e().eq(contents, PINNED_VERSION);
+}
+
+#[cargo_test]
+fn run_embed() {
+    const ECHO_SCRIPT: &str = r#"#!/usr/bin/env cargo
+
+fn main() {
+    let mut args = std::env::args_os();
+    let bin = args.next().unwrap().to_str().unwrap().to_owned();
+    let args = args.collect::<Vec<_>>();
+    println!("bin: {bin}");
+    println!("args: {args:?}");
+}
+
+#[test]
+fn test () {}
+"#;
+    let lockfile_path_argument = "mylockfile/Cargo.lock";
+    let p = project().file("src/main.rs", ECHO_SCRIPT).build();
```

Comment:
I guess we don't need any content inside the main function, since asserting them is the job of `script.rs` test suite.

```suggestion
    let p = project().file("src/main.rs", "fn main()").build();
```